### DISCUSSION TITLE
Select component. How to get the whole option as value in an onChange handler?

### DISCUSSION THREAD

**Discussion by yersan** [ant-design/ant-design#48297] [Q&A] [answered]
Hello, 

I have the following Select component:

```
<Select
  defaultValue="small"
  options={[
      {value: 'small', label: 'Small ' + data.priceSmall + ' €', price: data.priceSmall},
      {value: 'medium', label: 'Medium ' + data.priceMedium + ' €', price: data.priceMedium},
      {value: 'large', label: 'Large ' + data.priceLarge + ' €', price: data.priceLarge},
  ]}
  optionLabelProp="label"
    onChange={handleOnChangeSelectSize}
/>
```

And this is my onChange handler:

```
 const handleOnChangeSelectSize = (value) => {
        console.log(value);
    }
```

When I change the select value, I only get as `value` the string of the selected select value, I mean, depending on what I have selected I would get 'small', 'medium' or 'large'.

How I can get the whole option instead of only the value field?, I mean, If I've selected 'small', I would like to get the following option element of the options array: ` {value: 'small', label: 'Small ' + data.priceSmall + ' €', price: data.priceSmall}`

Thanks in advance

**Comment by kiner-tang** [comment, marked as answer]
You can get the selected option by this way:
```tsx
const handleOnChangeSelectSize = (value, option) => {
    console.log(value, option);
}
```

**Reply by yersan** [reply]
Worked like a charm, thanks @kiner-tang
